app/javascript/components/components/NewComponentModal.vue

```diff
@@ -133,46 +167,69 @@ export default {
     },
     createComponent: function (bvModalEvt) {
       this.loading = true;
+      let failed = false;
       bvModalEvt.preventDefault();
 
       // Guard before POST
-      if (!this.prefix) {
+      if (!this.prefix && !this.spreadsheet_import) {
         this.$bvToast.toast("Please enter a prefix", {
           title: "Error",
           variant: "danger",
           solid: true,
         });
-        return;
+        failed = true;
+      }
+      if (!this.file && this.spreadsheet_import) {
+        this.$bvToast.toast("Please select a spreadsheet to import", {
+          title: "Error",
+          variant: "danger",
+          solid: true,
+        });
+        failed = true;
       }
       if (!this.security_requirements_guide_id) {
         this.$bvToast.toast("Please select an SRG", {
           title: "Error",
           variant: "danger",
           solid: true,
         });
-        return;
+        failed = true;
       }
       if (!this.version) {
         this.$bvToast.toast("Please enter a version", {
           title: "Error",
           variant: "danger",
           solid: true,
         });
+        failed = true;
+      }
+      if (failed) {
+        this.loading = false;
```

Comment:
this works, but should there be an error message that pops up as well if there's an error other than `422` (like `500` from uploading a non excel file or something)

Reply:
I added some code to handle this, let me know you think.